### PR TITLE
Knative 0.17 images

### PR DESCRIPTION
**Pull request by matzew**
This bumps all the things to 0.17 

I have described on each commit, what the heck I did there for bumping

### CODE REVIEW

**Comment by matzew** [comment]
some panic on the upgrade:

```
NumActivators=3
panic: test timed out after 20m0s

goroutine 242 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1509 +0x11c
created by time.goFunc
	/usr/local/go/src/time/sleep.go:168 +0x52

goroutine 1 [chan receive, 20 minutes]:
testing.(*T).Run(0xc0001d85a0, 0x274ecc7, 0x9, 0x280f490, 0x0)
	/usr/local/go/src/testing/testing.go:1091 +0x739
testing.runTests.func1(0xc0001d85a0)
	/usr/local/go/src/testing/testing.go:1334 +0xa7
testing.tRunner(0xc0001d85a0, 0xc0003efd38)
	/usr/local/go/src/testing/testing.go:1039 +0x1ec
testing.runTests(0xc0007117c0, 0x3c96150, 0x1, 0x1, 0x0)
	/usr/local/go/src/testing/testing.go:1332 +0x528
testing.(*M).Run(0xc000734180, 0x0)
	/usr/local/go/src/testing/testing.go:1249 +0x440
main.main()
	_testmain.go:44 +0x224

goroutine 19 [chan receive]:
k8s.io/klog.(*loggingT).flushDaemon(0x3cbeb20)
	/go/src/knative.dev/serving/vendor/k8s.io/klog/klog.go:1010 +0xae
created by k8s.io/klog.init.0
	/go/src/knative.dev/serving/vendor/k8s.io/klog/klog.go:411 +0x18f
```

/test 4.5-upgrade-tests-aws-ocp-45

**Comment by matzew** [comment]
Some fail on the upstream test:

* --- FAIL: TestInitScaleZero (0.18s)
* --- FAIL: TestActivatorHANonGraceful (60.55s)

not exactly sure... therefore, I just try, again:

/test 4.5-upstream-e2e-aws-ocp-45

**Comment by nak3** [comment]
`TestInitScaleZero` needs enable `allow-zero-initial-scale` on `config-autoscaler` before running test. The nightly build enabled it as https://github.com/openshift/knative-serving/pull/521.
`TestActivatorHANonGraceful` also needs some tweak like https://github.com/openshift/knative-serving/pull/502

I will send the PR to your (@matzew) branch if you don't mind.

File: olm-catalog/serverless-operator/generate_csv.sh [hunk 1]
```diff
@@ -29,27 +29,21 @@ images=(
...
+  ["v0.17.0-pingsource-cleanup__pingsource"]="${eventing}-pingsource-cleanup"
   ["PING_IMAGE"]="${eventing}-ping"
   ["MT_PING_IMAGE"]="${eventing}-mtping"
   ["APISERVER_RA_IMAGE"]="${eventing}-apiserver-receive-adapter"
   ["BROKER_INGRESS_IMAGE"]="${eventing}-broker-ingress"
   ["BROKER_FILTER_IMAGE"]="${eventing}-broker-filter"
   ["DISPATCHER_IMAGE"]="${eventing}-channel-dispatcher"
 
-  ["KN_CLI_ARTIFACTS"]="${registry}/knative-v0.16.1:kn-cli-artifacts"
+  ["KN_CLI_ARTIFACTS"]="${registry}/knative-v0.17.1:kn-cli-artifacts"
```
Comment (markusthoemmes) [inline olm-catalog/serverless-operator/generate_csv.sh:46]: Just to doublecheck: Is this supposed to be 0.17.1?

File: olm-catalog/serverless-operator/generate_csv.sh [hunk 2]
```diff
@@ -29,27 +29,21 @@ images=(
...
-
-  # the mt broker replaces the removed broker
   ["mt-broker-controller__mt-broker-controller"]="${eventing}-mtchannel-broker"
   ["mt-broker-filter__filter"]="${eventing}-mtbroker-filter"
   ["mt-broker-ingress__ingress"]="${eventing}-mtbroker-ingress"
   ["imc-controller__controller"]="${eventing}-channel-controller"
   ["imc-dispatcher__dispatcher"]="${eventing}-channel-dispatcher"
 
-  ["v0.16.0-broker-cleanup__brokers"]="${eventing}-broker-cleanup"
+  ["v0.17.0-pingsource-cleanup__pingsource"]="${eventing}-pingsource-cleanup"
```
Comment (markusthoemmes) [inline olm-catalog/serverless-operator/generate_csv.sh:38]: Shouldn't this be 0.17.2?

File: olm-catalog/serverless-operator/generate_csv.sh (hunk 1 above)
Comment (matzew) [reply olm-catalog/serverless-operator/generate_csv.sh:46]: I looked here for the 0.17 bump:
https://github.com/openshift/release/blob/master/core-services/image-mirroring/knative/mapping_knative_v0_17_quay#L19

@navidshaikh or @rhuss  can you confirm ?

File: olm-catalog/serverless-operator/generate_csv.sh (hunk 2 above)
Comment (matzew) [reply olm-catalog/serverless-operator/generate_csv.sh:38]: Nope, unfortunately the name of the job is hard coded to `0.17.0` on the upstream `release-0.17` branch:
https://github.com/knative/eventing/blob/master/config/post-install/v0.17.0/pingsource-cleanup.yaml#L4

However, the image does contain the 0.17.2 build that we do:
https://github.com/openshift-knative/serverless-operator/pull/535/files#diff-f70d314a9f9d720fb31144a40e937449R299

@markusthoemmes I can change it in our operator fork to `v0.17` instead, if you feel like changing. :man_shrugging:

File: olm-catalog/serverless-operator/generate_csv.sh (hunk 1 above)
Comment (navidshaikh) [reply olm-catalog/serverless-operator/generate_csv.sh:46]: +1 
v0.17.1 is the latest

heads up: We're planning another patch release v0.17.2 upstream (today/tomorrow) for some requirements from faas team, we'd make sure to update this once the CI/mirroring is in place

Comment (matzew) [reply olm-catalog/serverless-operator/generate_csv.sh:46]: cool! Thanks for confirming. If needed we can either update this PR, or just bump as needed on a separate PR

**Comment by rhuss** [comment]
> I have described on each commit, what the heck I did there for bumping

cool, thanks ! I'm going to take that over into the documentation (along with your latest comments how to setup the CI, that I havent added yet).

File: olm-catalog/serverless-operator/generate_csv.sh (hunk 2 above)
Comment (markusthoemmes) [reply olm-catalog/serverless-operator/generate_csv.sh:38]: Got it, thanks. WFM if it's a mistake upstream anyway, we can keep it as-is.

**Review by markusthoemmes — APPROVED** [review]
/lgtm
/approve

**Comment by matzew** [comment]
```
/go/src/github.com/openshift-knative/serverless-operator/hack/lib/../../test/vendor/knative.dev/test-infra/scripts/library.sh: line 431: go-junit-report: command not found
```

is the error

**Comment by markusthoemmes** [comment]
The upgrade test failure is genuine.

The autoscaler added a new field to the KPAs that is read via the Revision and that field might not be propagated properly and/or might be racing. I'll retest to assert if it's indeed "just" a race or consistently failing but I don't quite know how this could've been passing upstream consistently yet. I'll provide a fix on the Serving side, should this be persistent (or actually: I'll provide a fix in any case).

/test 4.5-upgrade-tests-aws-ocp-45

**Comment by aliok** [comment]
/retest

```
+ echo DEBUG 14:07:18.788 'Execution failed: [[ $(oc get $kservice -n serving-tests -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") != True ]]. Waiting 5 sec (905/900)...'
```

File: olm-catalog/serverless-operator/generate_csv.sh (hunk 1 above)
Comment (navidshaikh) [reply olm-catalog/serverless-operator/generate_csv.sh:46]: kn v0.17.2 is ready, PTAL https://github.com/openshift-knative/serverless-operator/pull/544

**Pull request merged**